### ISSUE TITLE
overkilled

### ISSUE THREAD

**Issue by yfliao** [facebookresearch/denoiser#84]
Dear all,

For some speech files, the denoiser almost suppress all the speech signals in the utterance (as shown in the attached figure). Is there any threshold or option to avoid it?

Thanks for your help and have a nice day!

Best Regards,
yuanfu

![image](https://user-images.githubusercontent.com/6238892/121453760-3e46a900-c9d4-11eb-97d4-c5e710e7847d.png)

**Comment by stolpa4** [comment]
@yfliao Yes, I see these symptoms sometimes too. And what is strange, the SNR does not have to be really bad, speech can be quite loud, but it still will be suppressed. 

Denoiser's implementation is quite good, it is just the Demucs processes the signal that way.

I've experimented with other training data and loss parameters, but this behavior still persists. I will try to experiment more with it this summer.

There is actually one parameter in the `DemucsStreamer` class, called `dry` (its values must be from 0 to 1). It can be used to retain some part of the original signal in the result. 
It doesn't resolve the issue, as the signal in such cases will be severely attenuated, but you can use it to mitigate the problem depending on your use case.

**Comment by adiyoss** [comment]
Hi @yfliao,
Indeed, it can happen sometimes, are you using the `master64` model? 
In case you can share the audio file, we can take a deeper look at it.

**Comment by adefossez** [comment]
I added a change to automatically resample on the fly when necessary :) 
See https://github.com/facebookresearch/denoiser/commit/606d7695363422c7a474ab5d46f8a4f266b1e343
